lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3InputFile.java

```diff
@@ -130,7 +131,7 @@ private boolean headObject()
             return false;
         }
         catch (SdkException e) {
-            throw new IOException("S3 HEAD request failed for file: " + location, e);
+            throw unrecoverableIOException(new IOException("S3 HEAD request failed for file: " + location, e));
```

Comment:
Can we skip creating the `IOException` and only have the `UnrecoverableIOException`? The latter is a subclass of the former, and wrapping them will only make the stack longer.